File: 0003-longest-substring-without-repeating-characters/Solution.js
// Return the longest substring without repeating characters

/**
 * @param {string} s
 * @return {number}
 */

var lengthOfLongestSubstring = function(s) {
    if (s == "") { return 0; }

    let lengths = [];

    for (let x = 0; x < s.length; x++) {
        let temp = "";

        for (let z = x; z < s.length; z++) {
            let letter = s.charAt(z);

            if (temp.indexOf(letter) != -1) { break; }

            temp += letter;
        }

        lengths.push(temp.length);
    }

    lengths.sort((a, b) => { return a - b; });

    return lengths[lengths.length - 1];
};

File: 0003-longest-substring-without-repeating-characters/Solution.java
// Return the longest substring without repeating characters

class Solution {
    public int lengthOfLongestSubstring(String s) {
        if (s.length() == 0) {
            return 0;
        }

        List<Integer> lengths = new ArrayList<>();

        for (int x = 0; x < s.length(); x++) {
            StringBuilder temp = new StringBuilder("");

            for (int z = x; z < s.length(); z++) {
                String letter = String.valueOf(s.charAt(z));

                if (temp.indexOf(letter) != -1) {
                    break;
                }

                temp.append(letter);
            }

            lengths.add(temp.length());
        }

        Collections.sort(lengths);

        return lengths.get(lengths.size() - 1);
    }
}
